brownie/network/contract.py

```diff
@@ -775,8 +775,13 @@ def _check_and_set(self, name: str, obj: Any) -> None:
                 BrownieEnvironmentWarning,
             )
         elif hasattr(self, name):
-            raise AttributeError(f"Namespace collision: '{self._name}.{name}'")
-        setattr(self, name, obj)
+            warnings.warn(
+                f"WARNING: Namespace collision: '{self._name}.{name}'\n"
+                f"You will not be able to call _name() on {self._name}",
```

Comment:
```suggestion
                f"The {name} function will not be available when interacting with {self._name}",
```

Reply:
Generally not a fan of this but I understand why it's needed..  in a v2 need to solve this more elegantly.